lib/LoginStrategy.ts

```diff
@@ -75,10 +90,22 @@ export class LoginStrategy extends BaseStrategy {
   ) {
     super(options);
     this.claimField = claimField;
-    this.provider = new providers.JsonRpcProvider(rpcUrl);
-    this.ensResolver = PublicResolver__factory.connect(
-      ensResolverAddress,
-      this.provider
+    this.provider = new JsonRpcProvider(rpcUrl);
+    const { chainId, didRegistryAddress, ensRegistryAddress } =
+      chainConfigs()[rpcUrl];
```

Comment:
Instead of hardcoding the `chainId` based on `rpcUrl` could we instead get the `chainId` from the provider?

Reply:
Yes, but it will require initialization after construction because getting chain id is asynchronous. Can we allow this change?

Reply:
Hmm, how about this approach:
- Regardless of Volta/EWC, we add all of the Volta/EWC resolvers that we are aware of.
- We optionally allow specification of new resolvers in the constructor (as you have done), and also require the `chainId` for each resolver

Reply:
@jrhender Though we still need chain id to get ens and did registry addresses

Reply:
I am not sure. This way we introduce possibility for mismatch chain id and rpcUrl. It is more consistent to provide ens and DID registries along with resolvers

Reply:
> more consistent to provide ens and DID registries along with resolvers
Ok, this makes sense to me, good idea 👍